exchange/exchange.go

```diff
@@ -77,6 +82,18 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 	// Slice of BidRequests, each a copy of the original cleaned to only contain bidder data for the named bidder
 	blabels := make(map[openrtb_ext.BidderName]*pbsmetrics.AdapterLabels)
 	cleanRequests, aliases, errs := cleanOpenRTBRequests(bidRequest, usersyncs, blabels, labels)
+
+	// Clean PI from bidrequests if not allowed per GDPR
+	gdpr, consent, _ := extractGDPR(bidRequest, e.UsersyncIfAmbiguous)
+	// TODO : do something with the error value
+	if gdpr == 1 {
+		for bidder, bidReq := range cleanRequests {
+			if ok, err := e.gDPR.PersonalInfoAllowed(ctx, bidder, consent); !ok && err == nil {
```

Comment:
If `err != nil` here should we clean by default?

Reply:
perhaps if we have GDPR enabled when ambiguous. It should be rare that looking this up should generate an error.